CMakeLists.txt

```diff
@@ -14,7 +14,7 @@ cmake_minimum_required(VERSION 2.8...3.20 FATAL_ERROR)
 
 project(SOCI)
 
-set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD 17)
```

Comment:
I think we need to do
```cmake
if(NOT "${CMAKE_CXX_STANDARD}")
  set(CMAKE_CXX_STANDARD 14)
endif()
```
and add a CI build using `-DCMAKE_CXX_STANDARD=17` to test C++17 stuff.

Reply:
Apparently this doesn't work, sorry for proposing to do it without testing, see the failure in the clang build. It looks like `CMAKE_CXX_STANDARD` is always defined, so we need to check if it's `< 14` instead, apparently, and set it to 14 then.

We also still need a C++17 CI build as otherwise the new code is never tested.

Reply:
No Problem I can change the `CMAKE_CXX_STANDARD`.
But I have no idea how to setup a CI build for c++17.

Reply:
You can add C++17 build to `.github/workflows/ci.yml`.